packages/field-base/src/clear-button-mixin.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { dedupingMixin } from '@polymer/polymer/lib/utils/mixin.js';
+import { InputMixin } from './input-mixin.js';
+
+const ClearButtonMixinImplementation = (superclass) =>
+  class ClearButtonMixinClass extends InputMixin(superclass) {
+    static get properties() {
+      return {
+        /**
+         * Set to true to display the clear icon which clears the input.
+         * @attr {boolean} clear-button-visible
+         */
+        clearButtonVisible: {
+          type: Boolean,
+          reflectToAttribute: true,
+          value: false
+        }
+      };
+    }
+
+    /** @protected */
+    get _clearOnEsc() {
+      return true;
+    }
+
+    /** @protected */
+    ready() {
+      super.ready();
+
+      this.addEventListener('keydown', (e) => this._handleKeyDown(e));
+
+      this._clearElement = this.$.clearButton;
```

Comment:
The `DelegateFocusMixin` uses a getter that explicitely needs to be implemented to return the required element, and it gives a nice warning if not. I think this could be a good pattern in general for passing an element as a dependency to a mixin. Another alternative would be a `initClearButtonMixin` method and pass the dependencies through that.

Also this will result in a generic null reference error below if the element is not defined, which is different to how the `DelegateFocusMixin` handles elements.

Reply:
That's a good point. I have changed the implementation to align with the `DelegateFocusMixin` approach.